### PR TITLE
 Check variant, locus, interval coordinates consistent with reference

### PR DESCRIPTION
**Pull request by jigold**
Updated version of #2520. Will break import for some datasets!

### CODE REVIEW

**Comment by jigold** [comment]
@danking This is the same PR as before that you looked at. Unfortunately, github won't let me reopen the same PR once it has diverged from rebases.

File: src/test/scala/is/hail/io/LoadBgenSuite.scala [hunk 1]
```diff
@@ -264,7 +265,7 @@ class LoadBgenSuite extends SparkSuite {
 
   @Test def testReIterate() {
     hc.indexBgen("src/test/resources/example.v11.bgen")
-    val vds = hc.importBgen("src/test/resources/example.v11.bgen", Some("src/test/resources/example.sample"))
+    val vds = hc.importBgen("src/test/resources/example.v11.bgen", Some("src/test/resources/example.sample"), contigRecoding = Some(Map("01" -> "1")))
```
Comment (danking) [inline src/test/scala/is/hail/io/LoadBgenSuite.scala]: `contigRecoding = contigRecoding`?

**Comment by cseed** [comment]
bump.  needs rebase.

**Comment by jigold** [comment]
@danking Ready for review.

**Comment by danking** [comment]
@jigold more merge conflicts

**Pull request merged**